cli/src/commands/cloud/environments/tunnel/local.rs

```diff
@@ -271,105 +278,136 @@ where
             .await
             .map_err(ServeError::Connection)?;
 
-        #[allow(deprecated)]
-        hyper_0_14::server::conn::Http::new()
-            .serve_connection(io, &mut self)
-            .await?;
+        let this = Arc::new(RwLock::new(self));
```

Comment:
i dont think there is a problem because it will be read-locking only with no writers contending after the first setup and this is a cli feature anyway so we aren't going to expect thousands of concurrent invocations passing through it, i imagine

Reply:
does reducing the time that read locks is held meaningfully matter for anything? and the write lock being held for the duration of initialisation is indeed a goal, and really the main reason i am using a lock at all, because i dont want new requests coming in during that time (which was the previous behaviour)

Reply:
Yeah, I think acquiring the read lock will also only be an atomic increment or so. So ignore this comment.